packages/core/src/rum/instrumentation/resourceTracking/distributedTracing/__tests__/distributedTracing.test.ts

```diff
@@ -9,63 +17,327 @@ describe('TraceIdentifier', () => {
 
         // WHEN
         while (counter-- > 0) {
-            generatedIds.add(new TraceIdentifier().toString(10));
+            generatedIds.add(
+                TracingIdentifier.createTraceId().toString(
+                    TracingIdFormat.decimal
+                )
+            );
+            generatedIds.add(
+                TracingIdentifier.createSpanId().toString(
+                    TracingIdFormat.decimal
+                )
+            );
         }
 
         // THEN
-        expect(generatedIds.size).toBe(iterations);
+        expect(generatedIds.size).toBe(iterations * 2);
     });
 
-    it('M return an 64 bits positive integer W toString', async () => {
-        let iterations = 100;
-        while (iterations-- > 0) {
-            // GIVEN
-            const id = new TraceIdentifier().toString(10);
+    describe('Trace IDs', () => {
+        it('M return a valid 128 bits HEX string W toString(.hex)', async () => {
+            let iterations = 100;
+            while (iterations-- > 0) {
+                // GIVEN
+                const id = TracingIdentifier.createTraceId();
+                const idStr128 = id.toString(TracingIdFormat.hex);
 
-            // THEN
-            expect(id).toMatch(/[1-9]{1,19}/);
-            // should be less than the max 64 bits integer
-            if (id.length === 19) {
-                expect(id < '9223372036854775807').toBeTruthy();
+                // THEN
+                expect(idStr128).toMatch(/^[0-9a-f]{1,}$/);
```

Comment:
/improvement

Although, I would have preferred a simple check with string parsing which is easy, fast and maintainable however I would let you pick.

if it has to regex, let's make it more strict with something like
```
^[0-9a-f]{8}[0]{8}[0-9a-f]{16}$
```